bears/yang/YANGBear.py

```diff
@@ -0,0 +1,35 @@
+from os.path import dirname
+
+from coalib.bearlib.abstractions.Linter import linter
+from dependency_management.requirements.PipRequirement import PipRequirement
+
+
+@linter(executable='pyang',
+        use_stderr=True,
+        output_format='regex',
+        output_regex=r'.*:(?P<line>\d+):\s*(?P<message>.*)')
+class YANGBear:
+    """
+    Lints `YANG <https://en.wikipedia.org/wiki/YANG>`__ model files using
+    ``pyang``.
+
+    From `RFC 7950 <https://tools.ietf.org/html/rfc7950>`__:
+
+    _"YANG is a data modeling language used to model configuration data,
+    state data, Remote Procedure Calls, and notifications for network
+    management protocols."_
+    """
+    LANGUAGES = {'YANG'}
+    REQUIREMENTS = {PipRequirement('pyang', '1.7.1')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Formatting', 'Smell', 'Syntax'}
+    SEE_MORE = 'https://pypi.python.org/pypi/pyang'
+
+    @staticmethod
+    def create_arguments(filename, file, config_file):
+        # -p defines the search path for other YANG modules imported by
```

Comment:
@jayvdb I created a bear setting for defining additional search paths. This had anyway to be done. Is it ok to keep this code here under those new circumstances? If not, how to actually define a *"configuration_setting"*? `.generate_config()` and then reading the config from `config_file` in `.create_arguments()`? Since `pyang` doesn't support any config files for specifying cmdline options (if that's the actual intension of `config_file`)